### ISSUE TITLE
Checkpoints and random seeds

### ISSUE THREAD

**Issue by anands-repo** [pytorch/pytorch#26013]
**What I want to do:** I would like to read the current random generator state in pytorch. I know how to set the random generator state through manual_seed, but not how to read it. (assuming that this random generator state is the same as that which is initialized through torch.manual_seed). Could someone point me to this? I found out about torch.Generator, and can save the Generator's internal state, but I am not able to set torch.manual_seed using this state. 

**Use case:** When Dataloader uses shuffling or when SubsetRandomSampler is used with DataLoader, the batches that are presented to the DNN are dependent on the random number generator. It is possible that restoring from a checkpoint in the middle of a training iteration would present some examples twice to the DNN if the manual_seed is not set correctly to produce the same batches as before. (I couldn't find out how to pass a Generator object to the random sampler either ... )

I see a feature request: https://github.com/pytorch/pytorch/issues/11813, for saving DataLoader, which would be great. However, my question is different: just starting the iterator at a specific batch wouldn't be sufficient when random sampling is involved. Also, I can skip batches as needed in the iterator and save some time. However, without the ability to restore the random state of the shuffling algorithm in DataLoader/RandomSampler, it wouldn't be possible to correctly save and restore within a training iteration. My assumption is that that state is simply determined by the random generator state.

**Comment by soumith** [comment]
look for get_rng_state and set_rng_state in our docs: https://pytorch.org/docs/stable/search.html?q=get_rng_state&check_keywords=yes&area=default#
